Library/Types/CharType.php

```diff
@@ -50,7 +50,7 @@ public function toHex($caller, CompilationContext $compilationContext, Call $cal
     {
         $exprBuilder = BuilderFactory::getInstance();
         $functionCall = $exprBuilder->statements()
-            ->functionCall('sprintf', array($exprBuilder->literal(Types::STRING, '%X'), $caller))
+            ->functionCall('zephir_string_to_hex', array($caller))
```

Comment:
Comment about for this optimizer about method

> Transforms calls to method "toHex" to sprintf('%X') call

Looks like should be changed, or removed because it's pretty simple